### PR TITLE
Add Comet integration

### PR DESCRIPTION
**Pull request by AyushExel**
@glenn-jocher 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Integration of Comet ML callbacks into Ultralytics YOLOv8 for enhanced experiment tracking.

### 📊 Key Changes
- Added `comet.py` as a new module for handling Comet ML callbacks.
- Included Comet ML callbacks in `base.py` to be registered alongside existing integrations.
- Introduced functions to log hyperparameters, training metrics, model images, model details, and the best model to Comet ML at different stages of the training process.

### 🎯 Purpose & Impact
- **Enhanced Experiment Tracking:** Allows researchers and developers to automatically log and track various metrics and artifacts in Comet ML, providing a systematic way to monitor experiments.
- **Richer Analytics:** Logs not just the loss items but also model parameters, FLOPs, and inference speed for a comprehensive overview of the model performance.
- **Easier Model Management:** Trainers can directly log their best models for future reference or deployment within Comet ML’s platform.
- **Streamlined Workflow:** Integrating Comet ML with Ultralytics YOLO helps in maintaining a consistent workflow and could attract users already comfortable with Comet’s analytics framework.

👩‍💻 For developers and ML engineers, this integration means more powerful tracking and model management tools at their disposal, while 🌐 non-expert users benefit from a more intuitive way to visualize the performance of their object detection models.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #24366](https://github.com/ultralytics/ultralytics/pull/24366).

The PR removes the upper version constraint on `openvino` in the Conda CI environment while retaining installation through the `pytorch` and `conda-forge` channels alongside PyTorch, TorchVision, and Ultralytics. This enables CI coverage for newer OpenVINO releases, reduces maintenance from outdated version limits, and may improve compatibility with newer hardware and inference optimizations.

Thanks to [@onuralpszr](https://github.com/onuralpszr) for the implementation, with contributions from [@lakshanthad](https://github.com/lakshanthad).

To test the merged changes, install the latest development version:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Alternatively, you can wait for the next official PyPI release.

Please let us know whether these changes resolve the issue. Thank you 🙏 for reporting it, and any further feedback is welcome if the problem persists.
